lib/wasix/src/http/mod.rs

```diff
@@ -4,6 +4,12 @@ pub mod client_impl;
 #[cfg(feature = "host-reqwest")]
 pub mod reqwest;
 
+#[cfg(feature = "js")]
+mod web_http_client;
```

Comment:
Do you think it makes sense to isolate all the web/js specific code in an `arch/web/`?

I think that would be better for code structure instead of sprinkling web related code around the crate.

Reply:
I was actually thinking of doing a big re-structure of the `wasmer-wasix` crate at some point. At the moment, we've got a bunch of random stuff in random places and have made a lot of things public accidentally. For example, the `ModuleCache` and all that are stored under the `wasmer_wasix::runtime` namespace, which doesn't really make sense. 

It's a bit tricky because we have two orthogonal ways to organise things and neither one is obviously better than the other. One way is to split things up trait/component, while the other way is to split things up by architecture/target. 

I personally prefer to put trait implementations in their own module next to the trait they implement. It's just a case of making sure we work hard to keep the amount of conditional compilation under control.